src/main/groovy/edu/oregonstate/mist/permissionsapi/db/PermissionsDAO.groovy

```diff
@@ -0,0 +1,41 @@
+package edu.oregonstate.mist.permissionsapi.db
+
+import edu.oregonstate.mist.permissionsapi.core.Permissions
+import edu.oregonstate.mist.permissionsapi.mapper.PermissionsMapper
+
+import org.skife.jdbi.v2.sqlobject.Bind
+import org.skife.jdbi.v2.sqlobject.SqlQuery
+import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper
+
+@RegisterMapper(PermissionsMapper)
+interface PermissionsDAO extends Closeable {
+
+    @SqlQuery("""
+        SELECT OsuId,
+               Onid,
+               HumanResourcesSecurityPermissionLevel,
+               StudentSecurityPermissionLevel,
+               FinanceSecurityPermissionLevel
+               
+        FROM CoreDataWarehouse.sys2sys.CoreUserSecurityDomainPermissionLevels_EcsApi_1_1_0
+        WHERE OsuId = :id
+    """)
+    Permissions getPermissionsById(@Bind("id") String id)
+
+    @SqlQuery("""
+        SELECT OsuId,
+               Onid,
+               HumanResourcesSecurityPermissionLevel,
+               StudentSecurityPermissionLevel,
+               FinanceSecurityPermissionLevel
+               
+        FROM CoreDataWarehouse.sys2sys.CoreUserSecurityDomainPermissionLevels_EcsApi_1_1_0
+        WHERE OsuId = :id
+        OR Onid = :onid
+    """)
+    List<Permissions> getPermissions(@Bind("id") String id,
```

Comment:
Why does this return a list? I think you could safely remove one of these methods and just have one method to use for both endpoints.

Reply:
Should the GET /core-permissions endpoint return a single resource object then instead of an array? Would it also then return a 404 if no resource is found? I thought that it would have to return an array but it should only ever return one object or no objects.